codegen/facelift/facelift-codegen.py

```diff
@@ -320,8 +320,23 @@ def hasReturnValue(self):
 setattr(qface.idl.domain.Property, 'cppMethodArgumentType', property(cppMethodArgumentType))
 
 ##############################
-
-def generateFile(generator, outputPath, templatePath, context, libraryName, libraryType):
+def isContainsMocMacro(fileName, tmpPath):
+    mocMacro = ["Q_OBJECT", "Q_GADGET", "Q_NAMESPACE"]
+    with open('{0}/{1}'.format(tmpPath, fileName), 'r') as inFile:
+        for line in inFile:
+            if any(word in line for word in mocMacro):
+                return True
+    return False
+
+def addToNoMocList(fileName, tmpPath, path):
+    with open('{}/no_moc.cmake'.format(tmpPath), 'a') as inFile:
+         inFile.write('    {}/{}\n'.format(path, fileName))
+
+def addToNoMocIfContainsMacro(fileName, tmpPath, path):
+    if not isContainsMocMacro(fileName, tmpPath):
+        addToNoMocList(fileName, tmpPath, path)
+
+def generateFile(generator, tmpPath, templatePath, context, libraryName, libraryType,  ):
```

Comment:
generateFile should detect the nature of the generated file (whether it is a source, a header which needs to be MOCed or a header which does not require MOC) and adjust the lists of files so that the cmake file can be produced. That cmake file can be read by the facelft_add_interface function just after the code generator has been invoked.

Reply:
approach has been changed in the new version